ansible/modules/hashivault/hashivault_token_lookup.py

```diff
@@ -0,0 +1,78 @@
+
+#!/usr/bin/env python
+DOCUMENTATION = '''
+---
+module: hashivault_token_lookup
+version_added: "3.2.0"
+short_description: Hashicorp Vault token create module
+description:
+    - Module to create tokens in Hashicorp Vault.
+options:
+    url:
+        description:
+            - url for vault
+        default: to environment variable VAULT_ADDR
+    verify:
+        description:
+            - verify TLS certificate
+        default: to environment variable VAULT_SKIP_VERIFY
+    authtype:
+        description:
+            - "authentication type to use: token, userpass, github, ldap"
+        default: token
+    token:
+        description:
+            - token for vault
+        default: to environment variable VAULT_TOKEN
+    username:
+        description:
+            - username to login to vault.
+        default: False
+    password:
+        description:
+            - password to login to vault.
+        default: False
+    accessor:
+        description:
+            - If set, lookups will use the this accessor token
+    wrap_ttl:
+        description:
+            - Indicates that the response should be wrapped in a cubbyhole token with the requested TTL.
+'''
+EXAMPLES = '''
+---
+- hosts: localhost
+  tasks:
+      hashivault_token_lookup:
+        token: "{{vault_root_token}}"
+      register: "vault_token_lookup"
+'''
+
+
+def main():
+    argspec = hashivault_argspec()
+    argspec['accessor'] = dict(required=False, type='str')
+    argspec['wrap_ttl'] = dict(required=False, type='str')
+    module = hashivault_init(argspec)
+    result = hashivault_token_lookup(module.params)
+    if result.get('failed'):
+        module.fail_json(**result)
+    else:
+        module.exit_json(**result)
+
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.hashivault import *
+
+
+@hashiwrapper
+def hashivault_token_lookup(params):
+    client = hashivault_auth_client(params)
+    accessor = params.get('accessor')
```

Comment:
Okay, the only issue I see here is this is currently limited to only looking up the same token that was used for authentication.  I'm cool with limitation now and sometime in the future someone can add a lookup_token parameter or something to query on other ones.  I think aligning accessor as bool is a good call.